presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchRecordCursor.java

```diff
@@ -334,4 +274,87 @@ public Object getValue()
             return value;
         }
     }
+
+    /**
+     * Iterator to go over search results in a paged manner.
+     */
+    private static class SearchHitsIterator
+            extends AbstractIterator<SearchHit>
+    {
+        private final ElasticsearchQueryBuilder queryBuilder;
+        private final int maxHits;
+        private final Duration requestTimeout;
+        private final int maxAttempts;
+        private final Duration maxRetryTime;
+
+        private Iterator<SearchHit> searchHits;
+        private String scrollId;
+
+        SearchHitsIterator(ElasticsearchQueryBuilder queryBuilder, ElasticsearchConnectorConfig config)
+        {
+            this.queryBuilder = queryBuilder;
+            this.maxHits = config.getMaxHits();
+            this.requestTimeout = config.getRequestTimeout();
+            this.maxAttempts = config.getMaxRequestRetries();
+            this.maxRetryTime = config.getMaxRetryTime();
+        }
+
+        @Override
+        protected SearchHit computeNext()
+        {
+            if (scrollId == null) {
+                // make the first request and get the scroll id
+                SearchResponse response = getSearchResponse(queryBuilder);
+
+                if (response.getHits().getTotalHits() > maxHits) {
```

Comment:
The limit made sense when we were buffering in memory, but do we need this now that we’re scrolling through results?

Reply:
Makes sense. Just pushed a commit that removes the check. Also marked the config as `Deprecated` (not sure if this is the correct way to handle config props that are no longer needed)

Reply:
I'd prefer to remove this config, adding this on top of the class:
```java
@DefunctConfig({
        "elasticsearch.max-hits",
})
```

If there is a strong reason not to remove this now, then let's have a TODO here and cleanup later (soonish).

Reply:
Sure, added it to the `@DefunctConfig`.